u8timeseries/models/statistics.py

```diff
@@ -0,0 +1,136 @@
+from ..timeseries import TimeSeries
+import numpy as np
+from statsmodels.tsa.stattools import acf
+from scipy.stats import norm
+from statsmodels.tsa.seasonal import seasonal_decompose
+import matplotlib.pyplot as plt
+from typing import Tuple
+import math
+
+
+def check_seasonality(ts: 'TimeSeries', m: int = None, max_lag: int = 24, alpha: float = 5):
+    """
+    Returns whether the TimeSeries [ts] is seasonal with period [m] or not.
+
+    If [m] is None, the seasonality period is inferred from the Auto-correlation Function (ACF).
+
+    :param ts: The TimeSeries to check for seasonality.
+    :param m: The seasonality period to check.
+    :param max_lag: The maximal lag allowed in the ACF.
+    :param alpha: The desired confidence level (default 5%).
+    :return: A tuple (season, m), where s is a boolean indicating whether the TimeSeries has seasonality or not
+             and m is the seasonality period.
+    """
+    if m is not None:
+        if m < 0:
+            raise ValueError('m must be a positive integer.')
+
+    n_unique = np.unique(ts.values()).shape[0]
+
+    if n_unique > 1:  # Check for non-constant TimeSeries
+
+        r = acf(ts.values(), nlags=max_lag)  # In case user wants to check for seasonality higher than 24 steps.
+
+        grad = np.gradient(r)
+        signs_changes = np.diff(np.sign(grad))
+
+        if m is None:
+            # Tries to infer seasonality from AutoCorrelation Function if no value of m has been provided.
+            # We look for the first positive local maximum of the ACF by checking the sign changes in the gradient.
+
+            # Local maximum is indicated by signs_change == -2
+            if len(np.nonzero((signs_changes == -2))[0]) > 0:
+                m = np.nonzero((signs_changes == -2))[0][0] + 1
+            else:
+                return False, 0
+
+        # Check for local maximum when m is user defined
+        if np.nonzero((signs_changes == -2))[0][0] != m-1:
```

Comment:
So here you are testing only for the "shortest" period, right? There could be another, larger, period `m` if I'm correct?

Reply:
Yes, for a seasonality of order m, you also get a seasonality of order k*m (k >=2, though they may not be significant).
This also somehow ties in with the comment above pointing out that the first local maximum may not be significant and we should check further.

Reply:
Do we agree (my assumption) that if ACF at lag `m` is significant and that ACF at lag `2m` is also significant and even higher, we only consider `m` as seasonality right? 
In other words, we do not take the highest autocorrelation peak but the smallest significant lag as seasonality

Reply:
seasonalities need not be multiples of each others, right? So you could have a first peak at `m`, and then a larger peak at `1.5*m` for instance. In this case I would (intuitively) expect the method to return the largest peak.

Reply:
@Bugdup Yes! That is my understanding too. The seasonality order should explain all spikes, thus we consider the minimal lag that fits all the conditions.
You can think, as an example, of tides along a beach. There are two high tides and two low tides a day, and they sometimes are big tides ("grandes marées"). However any tide is relevant for fishermen :)

Reply:
@hrzn Let's say that we have two local maximums. Then I would expect their gcd to also provide a local significant local maximum.

Reply:
@hrzn Yes, I just picked this specific example actually, did not see the specificity

Reply:
I guess this depends right? You could have time series with two seasonalities. Imagine for instance (in a not-far-fetch-at-all example) the time series describing the photovoltaic production on a planet with two suns that are out of lock :) So in general you'll have peaks at multiples of the first peak (e.g. `1m`, `2m`, etc), but you can also have other peaks that are not multiple of the first ones (`3.1421m`, 6.2842m`, etc). Right?
Then I think which peak we choose is a matter of convention. The first local maximum has the advantage of being simple & intuitive :)

Reply:
Ok. I think I (finally) have a satisfying answer :)
I modeled your example above using sine functions with different periods (3 and 5) and add them together, tried some stuff, and here is my conclusion:

- What has been implemented right now "won't work" if there are multiple seasonality in the sense that we have to make a choice. So it should be used only under the assumption that the seasonality is unique.

- There are two reasonable ways to extend the inferring part of the seasonality check that I can think of right now. If there are two distinct seasonality period (i.e. one is not a multiple of the other), then:
    -  either set the global seasonality to their LCM.
    -  or use Fourier Analysis to separate the two frequencies and treat them separately.